### PR TITLE
Racastur/associations

### PR DESCRIPTION
**Pull request by ramacasturi**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [x] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/customproviders.json [hunk 1]
```diff
@@ -380,6 +380,225 @@
           "application/json"
         ]
       }
+    },
+    "/{scope}/providers/Microsoft.CustomProviders/associations/{associationName}": {
```
Comment (amarzavery) [inline specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/customproviders.json]: indentation 2 spaces, please.

File: specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/examples/createOrUpdateAssociation.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (amarzavery) [inline specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/examples/createOrUpdateAssociation.json]: indentation 2 spaces

File: specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/examples/createOrUpdateAssociation.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        },
+        "200": {
+            "headers": {},
+            "body": {
+                "name": "associationName",
+                "id": "/scope/providers/Microsoft.CustomProviders/associations/associationName",
+                "type": "Microsoft.CustomProviders/associations",
+                "properties": {
+                    "targetResourceId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/appRG/providers/Microsoft.Solutions/applications/applicationName",
+                    "provisioningState": "Accepted"
```
Comment (amarzavery) [inline specification/customproviders/resource-manager/Microsoft.CustomProviders/preview/2018-09-01-preview/examples/createOrUpdateAssociation.json]: provisioningState for 200 should be "Succeeded"

**Review by amarzavery — CHANGES_REQUESTED** [review]

**Review by amarzavery — APPROVED** [review]

**Comment by ravbhatnagar** [comment]
@sanjaiganesh - This has been merged already but service team wants an ARM review. Can you please take a look?

**Pull request merged**